lab/src/raft/leader_election.go

```diff
@@ -0,0 +1,33 @@
+package raft
+
+import (
+	"time"
+)
+
+func (rf *Raft) toCandidate() {
+	rf.state = CANDIDATE
+	rf.currentTerm++  // Increment currentTerm
+	rf.votedFor = rf.me  // Vote for self
+}
+
+func (rf *Raft) runLeaderElection() {
+	rf.toCandidate()
+
+	replies := make([]RequestVoteReply, len(rf.peers))
+	voteChan := make(chan int, len(rf.peers) - 1)
+
+	// TODO(ling): Implement vote
+}
+
+// Kick off leader election periodically by sending out RequestVote RPCs
+// when it hasn't heard from another peer for a while.
+func (rf *Raft) startLeaderElection() {
```

Comment:
Note that this should keep checking runtime in a loop whenever it is timeout or not

Reply:
Done.